server/controllers/finance/generalLedger/index.js

```diff
@@ -42,8 +46,46 @@ function list(req, res, next) {
     `;
 
   db.exec(sql)
-    .then(function (rows) {
-        res.status(200).json(rows);
+    .then((rows) => {
+      res.status(200).json(rows);
     })
     .catch(next);
 }
+
+/**
+ * GET /general_ledger/accounts
+ * list accounts and their solds
+ */
+function listAccounts(req, res, next) {
+  getlistAccounts()
+  .then((rows) => {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+ * @function getlistAccounts
+ * get list of accounts
+ */
+function getlistAccounts() {
+  const sql =
+    `SELECT aggregator.id, aggregator.number, aggregator.label,
+      IF(aggregator.balance >= 0, aggregator.balance, 0) AS debtor_sold,
+      IF(aggregator.balance < 0, -1 * aggregator.balance, 0) AS creditor_sold
+    FROM (
+      SELECT SUM(gl.debit_equiv - gl.credit_equiv) AS balance, 
+        a.id, a.number, a.label 
+      FROM general_ledger AS gl 
+        JOIN account a ON a.id = gl.account_id
+        JOIN project pro ON pro.id = gl.project_id
+        JOIN period per ON per.id = gl.period_id
+        JOIN user u ON u.id = gl.user_id 
```

Comment:
Do you really need all these JOINs?  It looks like you actually only need the `general_ledger` and the `account` table.